federation/pkg/kubefed/init/init.go

```diff
@@ -485,6 +507,21 @@ func createAPIServer(clientset *client.Clientset, namespace, name, image, creden
 		return dep, nil
 	}
 
+	if !cloudEnvironment {
```

Comment:
Defining pvc config in the spec and then removing it here if necessary may complicate future maintenance.  Consider excluding the pvc configuration from the spec and adding it in this block instead.